configure.ac

```diff
@@ -3393,15 +3424,18 @@ AC_ARG_ENABLE([sha],
     [ ENABLED_SHA=yes ]
     )
 
-if test "$ENABLED_SHA" = "no"
+if  test "$ENABLED_WOLFENGINE" = "no"
```

Comment:
Change this to:

```
if  test "$ENABLED_WOLFENGINE" = "no"
then
    ENABLED_SHA=no
fi

if test "$ENABLED_SHA" = "no"
then
    AM_CFLAGS="$AM_CFLAGS -DNO_SHA -DNO_OLD_TLS"
else
    # turn off SHA if leanpsk or leantls on
    if test "$ENABLED_LEANPSK" = "yes" || test "$ENABLED_LEANTLS" = "yes"
    then
        AM_CFLAGS="$AM_CFLAGS -DNO_SHA -DNO_OLD_TLS"
        ENABLED_SHA=no
    fi
fi
```

As it stands now, if the user does `--enable-engine --disable-sha`, the report at the end of the configure run will report no SHA support, but that isn't actually reflected in options.h (i.e. `#define NO_SHA` will NOT be there).

Reply:
hmmm....Made this change and commit hook tests are failing. I guess I need to further investigate.